iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -346,6 +346,58 @@ expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(ValueType&
     return *this;
 }
 
+template <typename ValueType, typename ErrorType>
+template <typename Optional, typename std::enable_if<is_optional<Optional>::value, int>::type>
+inline const expected<ValueType, ErrorType>&
+expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(typename Optional::type&)>& callable) const
+    noexcept
+{
+    return const_cast<expected*>(this)->and_then(callable);
+}
+
+template <typename ValueType, typename ErrorType>
+template <typename Optional, typename std::enable_if<is_optional<Optional>::value, int>::type>
+inline expected<ValueType, ErrorType>&
+expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(typename Optional::type&)>& callable) noexcept
+{
+    if (!this->has_error())
```

Comment:
Can we add a has_value function to the expected in this pull request? I know that it is not strictly needed, but I would prefer to have both (has_error and has_value) or only has_value for consistency.

If we do so, we could use it here.

Reply:
Will get it in a follow-up.